[jit][edge] Migrate ListType to DynamicType on mobile.

zhxchen17 opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #70509
* #70233
* __->__ #70212
* #70205

Use DynamicType instead of ListType all over the place in Lite Interpreter. Namely we need to modify the following places:
1. Type parser which produces the Type constants.
2. IValue::type() which returns reflected Type from IValues.
3. Helper functions to construct the container value.
4. Typechecks which test whether a type instance is a particular container type.

Differential Revision: [D33176931](https://our.internmc.facebook.com/intern/diff/D33176931/)

zhxchen17 committed:
[jit][edge] Migrate ListType to DynamicType on mobile.

Use DynamicType instead of ListType all over the place in Lite Interpreter. Namely we need to modify the following places:
1. Type parser which produces the Type constants.
2. IValue::type() which returns reflected Type from IValues.
3. Helper functions to construct the container value.
4. Typechecks which test whether a type instance is a particular container type.

Differential Revision: [D33176931](https://our.internmc.facebook.com/intern/diff/D33176931/)

[ghstack-poisoned]

zhxchen17 committed:
Update on "[jit][edge] Migrate ListType to DynamicType on mobile."


Use DynamicType instead of ListType all over the place in Lite Interpreter. Namely we need to modify the following places:
1. Type parser which produces the Type constants.
2. IValue::type() which returns reflected Type from IValues.
3. Helper functions to construct the container value.
4. Typechecks which test whether a type instance is a particular container type.

Differential Revision: [D33176931](https://our.internmc.facebook.com/intern/diff/D33176931/)

[ghstack-poisoned]

zhxchen17 committed:
Update on "[jit][edge] Migrate ListType to DynamicType on mobile."


Use DynamicType instead of ListType all over the place in Lite Interpreter. Namely we need to modify the following places:
1. Type parser which produces the Type constants.
2. IValue::type() which returns reflected Type from IValues.
3. Helper functions to construct the container value.
4. Typechecks which test whether a type instance is a particular container type.

Differential Revision: [D33176931](https://our.internmc.facebook.com/intern/diff/D33176931/)

[ghstack-poisoned]

zhxchen17 committed:
Update on "[jit][edge] Migrate ListType to DynamicType on mobile."


Use DynamicType instead of ListType all over the place in Lite Interpreter. Namely we need to modify the following places:
1. Type parser which produces the Type constants.
2. IValue::type() which returns reflected Type from IValues.
3. Helper functions to construct the container value.
4. Typechecks which test whether a type instance is a particular container type.

Differential Revision: [D33176931](https://our.internmc.facebook.com/intern/diff/D33176931/)

[ghstack-poisoned]

zhxchen17 committed:
Update on "[jit][edge] Migrate ListType to DynamicType on mobile."


Use DynamicType instead of ListType all over the place in Lite Interpreter. Namely we need to modify the following places:
1. Type parser which produces the Type constants.
2. IValue::type() which returns reflected Type from IValues.
3. Helper functions to construct the container value.
4. Typechecks which test whether a type instance is a particular container type.

Differential Revision: [D33176931](https://our.internmc.facebook.com/intern/diff/D33176931/)

[ghstack-poisoned]

zhxchen17 committed:
Update on "[jit][edge] Migrate ListType to DynamicType on mobile."


Use DynamicType instead of ListType all over the place in Lite Interpreter. Namely we need to modify the following places:
1. Type parser which produces the Type constants.
2. IValue::type() which returns reflected Type from IValues.
3. Helper functions to construct the container value.
4. Typechecks which test whether a type instance is a particular container type.

Differential Revision: [D33176931](https://our.internmc.facebook.com/intern/diff/D33176931/)

[ghstack-poisoned]

zhxchen17 committed:
Update on "[jit][edge] Migrate ListType to DynamicType on mobile."


Use DynamicType instead of ListType all over the place in Lite Interpreter. Namely we need to modify the following places:
1. Type parser which produces the Type constants.
2. IValue::type() which returns reflected Type from IValues.
3. Helper functions to construct the container value.
4. Typechecks which test whether a type instance is a particular container type.

Differential Revision: [D33176931](https://our.internmc.facebook.com/intern/diff/D33176931/)

[ghstack-poisoned]

zhxchen17 committed:
Update on "[jit][edge] Migrate ListType to DynamicType on mobile."


Use DynamicType instead of ListType all over the place in Lite Interpreter. Namely we need to modify the following places:
1. Type parser which produces the Type constants.
2. IValue::type() which returns reflected Type from IValues.
3. Helper functions to construct the container value.
4. Typechecks which test whether a type instance is a particular container type.

Differential Revision: [D33176931](https://our.internmc.facebook.com/intern/diff/D33176931/)

[ghstack-poisoned]

zhxchen17 committed:
Update on "[jit][edge] Migrate ListType to DynamicType on mobile."


Use DynamicType instead of ListType all over the place in Lite Interpreter. Namely we need to modify the following places:
1. Type parser which produces the Type constants.
2. IValue::type() which returns reflected Type from IValues.
3. Helper functions to construct the container value.
4. Typechecks which test whether a type instance is a particular container type.

Differential Revision: [D33176931](https://our.internmc.facebook.com/intern/diff/D33176931/)

[ghstack-poisoned]

zhxchen17 committed:
Update on "[jit][edge] Migrate ListType to DynamicType on mobile."


Use DynamicType instead of ListType all over the place in Lite Interpreter. Namely we need to modify the following places:
1. Type parser which produces the Type constants.
2. IValue::type() which returns reflected Type from IValues.
3. Helper functions to construct the container value.
4. Typechecks which test whether a type instance is a particular container type.

Differential Revision: [D33176931](https://our.internmc.facebook.com/intern/diff/D33176931/)

[ghstack-poisoned]

zhxchen17 committed:
Update on "[jit][edge] Migrate ListType to DynamicType on mobile."


Use DynamicType instead of ListType all over the place in Lite Interpreter. Namely we need to modify the following places:
1. Type parser which produces the Type constants.
2. IValue::type() which returns reflected Type from IValues.
3. Helper functions to construct the container value.
4. Typechecks which test whether a type instance is a particular container type.

Differential Revision: [D33176931](https://our.internmc.facebook.com/intern/diff/D33176931/)

[ghstack-poisoned]

zhxchen17 committed:
Update on "[jit][edge] Migrate ListType to DynamicType on mobile."


Use DynamicType instead of ListType all over the place in Lite Interpreter. Namely we need to modify the following places:
1. Type parser which produces the Type constants.
2. IValue::type() which returns reflected Type from IValues.
3. Helper functions to construct the container value.
4. Typechecks which test whether a type instance is a particular container type.

Differential Revision: [D33176931](https://our.internmc.facebook.com/intern/diff/D33176931/)

[ghstack-poisoned]

iseeyuan reviewed (commented):
Could you verify if ListType is covered by a test? I could see Dict and Tuple tests here: https://github.com/pytorch/pytorch/blob/master/test/cpp/jit/test_lite_interpreter.cpp#L171-L205. The way they are tested may not be perfect. Feel free to use better way if there is any. If not covered, please add corresponding tests.

zhxchen17 committed:
Update on "[jit][edge] Migrate ListType to DynamicType on mobile."


Use DynamicType instead of ListType all over the place in Lite Interpreter. Namely we need to modify the following places:
1. Type parser which produces the Type constants.
2. IValue::type() which returns reflected Type from IValues.
3. Helper functions to construct the container value.
4. Typechecks which test whether a type instance is a particular container type.

Differential Revision: [D33176931](https://our.internmc.facebook.com/intern/diff/D33176931/)

[ghstack-poisoned]

zhxchen17 commented:
Added a test.

iseeyuan reviewed (approved):
LGTM!

zhxchen17 committed:
Update on "[jit][edge] Migrate ListType to DynamicType on mobile."


Use DynamicType instead of ListType all over the place in Lite Interpreter. Namely we need to modify the following places:
1. Type parser which produces the Type constants.
2. IValue::type() which returns reflected Type from IValues.
3. Helper functions to construct the container value.
4. Typechecks which test whether a type instance is a particular container type.

Differential Revision: [D33176931](https://our.internmc.facebook.com/intern/diff/D33176931/)

[ghstack-poisoned]

zhxchen17 committed:
Update on "[jit][edge] Migrate ListType to DynamicType on mobile."


Use DynamicType instead of ListType all over the place in Lite Interpreter. Namely we need to modify the following places:
1. Type parser which produces the Type constants.
2. IValue::type() which returns reflected Type from IValues.
3. Helper functions to construct the container value.
4. Typechecks which test whether a type instance is a particular container type.

Differential Revision: [D33176931](https://our.internmc.facebook.com/intern/diff/D33176931/)

[ghstack-poisoned]

zhxchen17 committed:
Update on "[jit][edge] Migrate ListType to DynamicType on mobile."


Use DynamicType instead of ListType all over the place in Lite Interpreter. Namely we need to modify the following places:
1. Type parser which produces the Type constants.
2. IValue::type() which returns reflected Type from IValues.
3. Helper functions to construct the container value.
4. Typechecks which test whether a type instance is a particular container type.

Differential Revision: [D33176931](https://our.internmc.facebook.com/intern/diff/D33176931/)

[ghstack-poisoned]

zhxchen17 committed:
Update on "[jit][edge] Migrate ListType to DynamicType on mobile."


Use DynamicType instead of ListType all over the place in Lite Interpreter. Namely we need to modify the following places:
1. Type parser which produces the Type constants.
2. IValue::type() which returns reflected Type from IValues.
3. Helper functions to construct the container value.
4. Typechecks which test whether a type instance is a particular container type.

Differential Revision: [D33176931](https://our.internmc.facebook.com/intern/diff/D33176931/)

[ghstack-poisoned]

zhxchen17 committed:
Update on "[jit][edge] Migrate ListType to DynamicType on mobile."


Use DynamicType instead of ListType all over the place in Lite Interpreter. Namely we need to modify the following places:
1. Type parser which produces the Type constants.
2. IValue::type() which returns reflected Type from IValues.
3. Helper functions to construct the container value.
4. Typechecks which test whether a type instance is a particular container type.

Differential Revision: [D33176931](https://our.internmc.facebook.com/intern/diff/D33176931/)

[ghstack-poisoned]

zhxchen17 committed:
Update on "[jit][edge] Migrate ListType to DynamicType on mobile."


Use DynamicType instead of ListType all over the place in Lite Interpreter. Namely we need to modify the following places:
1. Type parser which produces the Type constants.
2. IValue::type() which returns reflected Type from IValues.
3. Helper functions to construct the container value.
4. Typechecks which test whether a type instance is a particular container type.

Differential Revision: [D33176931](https://our.internmc.facebook.com/intern/diff/D33176931/)

[ghstack-poisoned]
